src/sidebar/util/time.ts

```diff
@@ -1,272 +1,12 @@
-    return null;
-  }
-
-  // We don't want to refresh anything more often than 5 seconds
-  nextUpdate = Math.max(nextUpdate, 5 * SECOND);
-
-  // setTimeout limit is MAX_INT32=(2^31-1) (in ms),
-  // which is about 24.8 days. So we don't set up any timeouts
-  // longer than 24 days, that is, 2073600 seconds.
-  nextUpdate = Math.min(nextUpdate, 2073600 * SECOND);
-
-  return nextUpdate;
-}
-
-/**
- * Start an interval whose frequency depends on the age of a timestamp.
- *
- * This is useful for refreshing UI components displaying timestamps generated
- * by `formatRelativeDate`, since the output changes less often for older timestamps.
- *
- * @param date - Date string to use to determine the interval frequency
- * @param callback - Interval callback
- * @return A function that cancels the interval
- */
-export function decayingInterval(
-  date: string,
-  callback: () => void,
-): () => void {
-  let timer: number | undefined;
-  const timestamp = new Date(date);
-
-  const update = () => {
-    const fuzzyUpdate = nextFuzzyUpdate(timestamp, new Date());
-    if (fuzzyUpdate === null) {
-      return;
-    }
-    const nextUpdate = fuzzyUpdate + 500;
-    timer = setTimeout(() => {
-      callback();
-      update();
-    }, nextUpdate);
-  };
-
-  update();
-
-  return () => clearTimeout(timer);
-}
-
-/**
- * Formats a date as a short approximate string relative to the current date.
- *
- * The level of precision is proportional to how recent the date is.
- *
- * For example:
- *
- *  - "Just now"
- *  - "5 minutes ago"
- *  - "25 Oct 2018"
- *
- * @param date - The date to consider as the timestamp to format.
- * @param now - The date to consider as the current time.
- * @param Intl - Test seam. JS `Intl` API implementation.
- * @return A 'fuzzy' string describing the relative age of the date.
- */
-export function formatRelativeDate(
-  date: Date | null,
-  now: Date,
-  Intl?: IntlType,
-): string {
-  if (!date) {
-    return '';
-  }
-  return getBreakpoint(date, now).formatter(date, now, Intl);
-}
-
-/**
- * Formats a date as an absolute string in a human readable format.
- *
- * The exact format will vary depending on the locale, but the verbosity will
- * be consistent across locales. In en-US for example this will look like:
- *
- *  "Sunday, Dec 17, 2017, 10:00 AM"
- *
- * @param Intl - Test seam. JS `Intl` API implementation.
- */
-export function formatDate(date: Date, Intl?: IntlType): string {
-  return format(
-    date,
-    {
-      year: 'numeric',
-      month: 'short',
-      day: '2-digit',
-      weekday: 'long',
-      hour: '2-digit',
-      minute: '2-digit',
-    },
-    Intl,
-  );
-}
-
 /**
  * Formats a date as `YYYY-MM-DD hh:mm`, using 24h and system timezone.
+ *
+ * This format has the next characteristics:
+ * - Lexicographic and chronological order match.
+ * - It is detected as "date" when pasted in common spreadsheet editors, making
+ *   it useful when exporting dates for CSV documents.
```

Comment:
I would say something like "It is detected and correctly parsed as a date". As well as being detected correctly, it matters that the parsing doesn't swap day and month for example. That could happen if pasting a DD-MM-YY date into an editor that expects MM-DD-YY.